### PR TITLE
Downgrade null coalescing operator

### LINKED ISSUE #4124: [Downgrade] PHP 7.4 - Null coalescing assignment operator
# Feature Request

Replace PHP 7.4's [null coalescing assignment operator `??`](https://www.php.net/manual/en/migration74.new-features.php#migration74.new-features.core.null-coalescing-assignment-operator) with its equivalent pre-7.4 code.
 
## Diff 

```diff
-$array['key'] ??= computeDefault();
+if (!isset($array['key'])) {
+    $array['key'] = computeDefault();
+}
```

### PR DESCRIPTION
**Pull request by leoloso**
Fixes #4124

### CODE REVIEW

File: rules/downgrade/tests/Rector/Coalesce/DowngradeNullCoalescingOperatorRector/Fixture/fixture.php.inc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Rector\Downgrade\Tests\Rector\Coalesce\DowngradeNullCoalescingOperatorRector\Fixture;
+
+$array = [];
+$array['user_id'] ??= 'value';
```
Comment (TomasVotruba) [inline rules/downgrade/tests/Rector/Coalesce/DowngradeNullCoalescingOperatorRector/Fixture/fixture.php.inc:6]: Could you wrap this into PHP class as other fixtures?
Now this code gets executed, as this file is included so test can use reflection and other static analysis for it

**Review by TomasVotruba — COMMENTED** [review]
Very nice work. One detail and it's ready to merge

File: rules/downgrade/tests/Rector/Coalesce/DowngradeNullCoalescingOperatorRector/Fixture/fixture.php.inc (hunk 1 above)
Comment (leoloso) [reply rules/downgrade/tests/Rector/Coalesce/DowngradeNullCoalescingOperatorRector/Fixture/fixture.php.inc]: Oops, I was not aware of the issue, fixed now

**Comment by TomasVotruba** [comment]
Failing phpstan is unrelated, I'll handle it later today

**Comment by TomasVotruba** [comment]
Thank you

**Pull request merged**